packages/api/lib/granule-remove-from-cmr.js

```diff
@@ -1,39 +1,37 @@
-const { GranuleNotPublished, RecordDoesNotExist } = require('@cumulus/errors');
+const { GranuleNotPublished } = require('@cumulus/errors');
 const { CMR } = require('@cumulus/cmr-client');
 const log = require('@cumulus/common/log');
-const { CollectionPgModel, GranulePgModel } = require('@cumulus/db');
+const { getGranuleCollectionId, GranulePgModel } = require('@cumulus/db');
```

Comment:
Were the changes here needed given we're doing the translation in the endpoint scope? 

Meaning we have an apiGranule that's derived from `translatePostgresGranuleToApiGranule` *not* dynamo, given this is an API lib, should we be using postgres objects here?

Reply:
Not sure I totally follow. The `unpublishGranule` function is now driven by a Postgres Granule instead of a Dynamo or API granule. I think this was brought in to get the collection ID from the PG Granule as that's not stored on the Granule directly.

That's a good question if we should not be using PG objects in api/lib. It seems like this function does need to use PG objects so I think the alternative would be moving `unpublishGranule`. Thinking through that... I don't know if there's an obviously better spot for it. I wouldn't want it in `db` I don't think. 

I'm leaning towards saying it's fine to use PG objects in api/lib. The `api` package will need to interact with PG and is no tightly coupled to Dynamo records. Eh?

Reply:
So *perhaps* that's fair enough that we should be assuming postgres object in api/lib, but... 

THe guts of the question was this - the endpoint method that uses this call, `put` from the granules endpoint is already doing the lookup in the parent method 

```
  const pgGranule = await getUniquePgGranuleByGranuleId(knex, granulePgModel, granuleId);
  const apiGranule = await translatePostgresGranuleToApiGranule(pgGranule, knex);
```

So the lookups from translation are adding another query to the database for information we already have in the parent method.   Initially I was asking if, given we don't *need* any lookups if an api granule is passed in, if this was necessary for an api-object oriented interface. 

 
However.... bulkGranuleDelete does not, which I missed (it's abstracted as a call to `unpublishGranuleFunc` :\ ) initially when looking at this.  

It's probably worth a sidebar re: our pattern in api libs but the main thrust of the concern is/was the duplicative postgres query.

Reply:
Made a few changes here to eliminate duplicative Collection queries:

https://github.com/nasa/cumulus/pull/2415/commits/3a817e513c4047a7a776a83c94347f69313673b6

This refactors the `unpublishGranule` and translate functions to accept an optional PG Collection. If passed, we'll skip the lookup in each of those. The one and only Collection query now comes from the `put` endpoint directly and that's passed around.

Reply:
I think that's a reasonable approach.